### PR TITLE
Feat/multimodal convert fix

### PR DESCRIPTION
**Pull request by Laz4rz**
Old convert script is not working for current SwissAI Megatron Multimodal branch, model fails to be loaded and saved at torch -> hf stage.

### CODE REVIEW

**Comment by Laz4rz** [comment]
This PR:
1. Adds missing qk layernom and xielu args reading from checkpoint in load_args_from_checkpoint
2. Adds missing rope args reading from checkpoint in load_args_from_checkpoint (@RaphaelKreft)
3. Adds handling of qk layernom and xielu tensors in saver
4. Switches convert.py to use serial loader/saver (happy to discuss more)
5. convert.py runs without torchrun and creates fake tp/ep groups to handle checks, this approach started failing and there are two different fixes for that proposed in [6c0216a](https://github.com/swiss-ai/Megatron-LM/pull/106/commits/6c0216a670f17e1c853b05ba1e92f9d9819ad860) and [1f67baa](https://github.com/swiss-ai/Megatron-LM/pull/106/commits/1f67baa9e4382a48df0843e45ab7e12b26934881) (very happy to discuss, this still feels patchworky)

Logits comparison for original torch dist checkpoint and hf checkpoints with all fixes and two different fake tp/ep groups fixes:
```
report_file                                         | mode        | timestamp_utc    | top5_ids                  | top5_logits                                       | logits_sha256
----------------------------------------------------+-------------+------------------+---------------------------+---------------------------------------------------+--------------
hf_general_tp_fix.report.json                       | hf          | 20260227T145722Z | 1010,1033,11330,1267,1032 | 16.750000,15.187500,15.062500,15.000000,14.687500 | f05a3f74f194
hf_tight_tp_fix.report.json                         | hf          | 20260227T152603Z | 1010,1033,11330,1267,1032 | 16.750000,15.187500,15.062500,15.000000,14.687500 | f05a3f74f194
native_tp_fix.report.json                           | native_dist | 20260227T153256Z | 1010,1033,11330,1267,1032 | 16.848190,15.274855,15.183476,15.014288,14.694513 | 7524c06ef499
```

**Comment by RaphaelKreft** [comment]
Added @AllenHaoHuang as he is mainly responsible for pretrain codebase

**Comment by RaphaelKreft** [comment]
As the logits matched in previous iterations as well: @Laz4rz did you check the output config.json in the hf cehckpoint to see whether the rope settings and all you expect are in there correctly? :)
